metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/configuration/RepositoryConfiguration.java

```diff
@@ -35,4 +36,9 @@ AnalysisEventHandler analysisEventHandler() {
         return new AnalysisEventHandler();
     }
 
+    @Bean
+    ReferenceSequenceEventHandler referenceSequenceEventHandler() {
```

Comment:
Do we really need the event handler to validate the taxonomy? Is `@ManyToOne(optional=false)` not enough?

Reply:
When invalid links are sent and/or patch happens this will not catch. Hence even handler is reqiured

Reply:
When invalid links are used (as in "Scenario: update a reference sequence with invalid taxonomy and fail"), the exception is caught, but not handled: https://github.com/EBIvariation/metadata-ws/blob/master/metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/exceptionhandling/ExceptionHandlers.java#L43

My point was if there is nothing this event handler does that other exception handler can't do then we should avoid duplication. We can't add event handler for each madantory link.

Reply:
It can be done. But we need to pass generic message to user.
Otherwise we need to parse strings, where the format might change with spring versions